Docker/FreshRSS.Apache.conf

```diff
@@ -1,5 +1,6 @@
 ServerName freshrss.localhost
 Listen 0.0.0.0:80
+Listen [::]:80
```

Comment:
Any advantage compared to a single `Listen 80` (which is the default Apache value)?
I am a bit worried about the risks of regressions by providing this line by default, e.g. for platforms without IPv6. At least we would need some tests to see how it behaves

Reply:
> which is the default Apache value

I thought Apache without a Listen directive won't start at all?

Reply:
> I thought Apache without a Listen directive won't start at all?

Apache does have a default Listen directive by default: `Listen 80`

Reply:
The single Listen 80 only accepts IPv4 connections. I couldn't find any other way except to make a docker volume for the config and edit it manually.
I don't know how Apache will behave on machines without IPv6 support. I think I could either make a separate variable like ENABLE_IPV6 or add Listen only when LISTEN6 is set

Reply:
> The single `Listen 80` only accepts IPv4 connections

Could you please double-check that?

 `Listen 80` works for me both for our Debian and our Alpine images, that is to say with IPv4 + IPv6. This is also what the documentation says https://httpd.apache.org/docs/current/mod/mpm_common.html#listen

```sh
docker run --rm -p 8080:80 -e 'LISTEN=80' -v freshrss_data:/var/www/FreshRSS/data freshrss/freshrss:edge
# or
docker run --rm -p 8080:80 -e 'LISTEN=80' -v freshrss_data:/var/www/FreshRSS/data freshrss/freshrss:edge-alpine

curl -v 'http://127.0.0.1:8080/'
curl -v 'http://[::1]:8080/'
```

Reply:
> Could you please double-check that?
Oh, my bad. The configuration was `Listen 0.0.0.0:80` and I didn't even try to do `Listen 80`. Yes, `Listen 0.0.0.0:80` only accepts IPv4, but `Listen 80` accepts both IPv4 and IPv6.
It's much simpler this way. I will make changes to leave the ability to bind to specific IPv6 addresses

Reply:
On an ancient Debian I can confirm that Listen 80 doesn't do that. On some version you should actually be using it works though. :-P
```
apache2/oldoldstable,now 2.4.25-3+deb9u13 amd64 [installed,automatic]
  Apache HTTP Server
```

```
/etc/apache2# grep -R Listen
ports.conf:Listen 80
ports.conf:     Listen 443
ports.conf:     Listen 443
```

```
curl -v 'http://127.0.0.1:80/'
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 80 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 200 OK
```

```
curl -v 'http://[::1]:80/'
*   Trying ::1...
* TCP_NODELAY set
* connect to ::1 port 80 failed: Connection refused
* Failed to connect to ::1 port 80: Connection refused
* Closing connection 0
curl: (7) Failed to connect to ::1 port 80: Connection refused
```

vs

```
ping ::1
PING ::1(::1) 56 data bytes
64 bytes from ::1: icmp_seq=1 ttl=64 time=0.042 ms
64 bytes from ::1: icmp_seq=2 ttl=64 time=0.054 ms
64 bytes from ::1: icmp_seq=3 ttl=64 time=0.050 ms
```

Reply:
Hmm. What do you think about making the LISTEN comma-separated variable a list of addresses/ports? This would allow you not to bind to just one address and allow you to bind to multiple addresses at once. But I think that would be a bit harder to implement

Reply:
As this is solely for our Docker images, we control Apache versions.
One thing at a time: could you please try again with the `Listen 80` approach? (Check my example above)

Reply:
Ah sorry, I did not see your answer...

Reply:
Given that `Listen 80` works, and with our existing environment variable, we already cover the ability to listen on IPv4, IPv6, or both, which I believe is sufficient for the context of being used in a container.
But we should at least update the documentation to explain how to do that.
We could also test some more to consider using `Listen 80` by default